python/orca/src/bigdl/orca/learn/pytorch/torch_runner.py

```diff
@@ -262,17 +270,26 @@ def train_epochs(self, data_creator, epochs=1, batch_size=32, profile=False,
                     self.logger.info(f"Finished training epoch {i + 1}, " +
                                      f"stats on rank 0: {stats}")
             stats_list.append(stats)
+            self.epochs_stats = stats
+            for callback in callbacks:
+                callback.on_epoch_end(epoch=self.epochs, logs=stats)
+        for callback in callbacks:
+            callback.on_train_end()
         return stats_list
 
     def train_epoch(self,
                     data_loader,
                     profile=False,
-                    info=None):
+                    info=None,
+                    callbacks=None):
         """Runs a training epoch and updates the model parameters."""
         if hasattr(self.train_loader, "sampler") and hasattr(
                 self.train_loader.sampler, "set_epoch"):
             self.train_loader.sampler.set_epoch(self.epochs)
         self.logger.debug("Begin Training Step {}".format(self.epochs + 1))
+        for callback in callbacks:
+            # TODO: train_stats
+            callback.on_batch_begin(self.epochs, logs=self.epoch_stats)
```

Comment:
why calling `on_batch_begin` here? they should be called at the beginning of every batch, right?